utilities/xmlrpcpp/src/XmlRpcServerConnection.cpp

```diff
@@ -335,6 +344,12 @@ XmlRpcServerConnection::generateResponse(std::string const& resultXml)
   std::string header = generateHeader(body);
 
   _response = header + body;
+  // Avoid an overly large response
+  if (_response.length() > __INT_MAX__) {
```

Comment:
In this case, the `_response()` length check is made after a potentially expensive string concatenation operation in the line above; the consistency of this block is very nice, but it might be better to avoid the string concatenation operation entirely if we're just going to discard the result. What would you think of checking `header.length() + body.length()` instead?

Reply:
Addressed in 31ab4b57f2362d95d4fa9772a0e5b1b1878ca747. Thanks, this sped up tests significantly!